src/app/loadout-builder/process-worker/process.ts

```diff
@@ -190,25 +190,22 @@ export function process({
               setStatistics.skipReasons.noExotic += 1;
               continue;
             }
-            let insufficientSetBonus = false;
-            for (const setHash of Object.keys(setBonuses)) {
-              // TODO unscrew
-              const setCount = [
-                helm.setBonus,
-                gaunt.setBonus,
-                chest.setBonus,
-                leg.setBonus,
-                classItem.setBonus,
-              ].filter((x) => x === Number(setHash)).length;
-              if (setCount < (setBonuses[Number(setHash)] || 0)) {
-                insufficientSetBonus = true;
-                break;
+
+            // Check set bonus requirements
+            // eslint-disable-next-line @typescript-eslint/prefer-for-of
+            for (let i = 0; i < setBonusHashes.length; i++) {
+              const setHash = setBonusHashes[i];
+              const setCount =
+                Number(helm.setBonus === setHash) +
+                Number(gaunt.setBonus === setHash) +
+                Number(chest.setBonus === setHash) +
+                Number(leg.setBonus === setHash) +
+                Number(classItem.setBonus === setHash);
+              if (setCount < setBonusCounts[setHash]) {
```

Comment:
How does setBonusCounts[setHash] function here if setBonusCounts is only the values of setBonuses?

Reply:
Ah yeah good carch